### PR TITLE
Build UI

### PR DESCRIPTION
**Pull request by Jaymelfah**
#### In this branch, I created a form of UI for the school library app. This way it can be invoked as an executable. 

> Created app.rb with various methods for the library app
> Created a main.rb file that defines the entry point of the app,  In this there is  a method called main that is invoked at the end of the file.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    if @book.empty?
+      puts 'No books available'
+      show_menu
+    end
+    puts 'List of available books:'
+    puts ''
+    @book.each_with_index do |item, i|
+      puts "#{i} Title: #{item.title}, Author: #{item.author}"
+    end
+  end
```
Comment (HAMMAS-SALEEM) [inline app.rb]: image

- [x] When i create a book and choose the option to see the books then after showing books your application automatically exits :thinking: Kindly fix this issue :+1:

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      puts 'No people found today'
+      show_menu
+    end
+    puts 'List of people:'
+    puts ''
+    @people.each_with_index do |item, i|
+      puts "#{i}-- [#{item.class}] Name:#{item.name},  Age: #{item.age}"
+    end
+  end
+
```
Comment (HAMMAS-SALEEM) [inline app.rb]: image

- [ ] When i create a person and choose the option to see the people then after showing people your application automatically exits 🤔 Kindly fix this issue :+1:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @Jaymelfah :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    list_books
+    index = gets.chomp.to_i
+    book = @book[index]
+    puts 'Select a person from the list (Enter number)'
+    list_people
+    person_index = gets.chomp.to_i
+    person = @people[person_index]
+    print 'Date(yyyy/mm/dd): '
+    date = gets.chomp
+    rental = Rental.new(date, book, person)
```
Comment (Davidosky007) [inline app.rb]: * Nice job working on the requested changes but seems while working on those changes you broke some parts of your app as rentals can no longer be created, Kindly fix this as before the changes they worked 🙏 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Jaymelfah 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 
 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @Jaymelfah , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights :green_circle: 

- All Linter checks passed :heavy_check_mark: 
- App function as expected :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**